docs/dev/reference/cbox.md

```diff
@@ -176,6 +183,93 @@ block data in `payload.content` is technically encoded four times:
 
 Not all commands require the request payload to be set or to include content. Not all commands include payload objects in their response.
 
+## Block IDs
+
+Blocks have both a 32-bit numeric ID and a string name.
+The numeric ID is considered the primary identifier, but the string name is required to be unique,
+and can be used as secondary (human-readable) ID.
+
+Request payloads are required to include either ID or name. If both are included, they must both refer to the same block.
+Response payloads will always include block ID, but may omit block name.
+There are two exceptions: responses to the `NAME_READ`, `NAME_READ_ALL`, and `NAME_WRITE` commands,
+and responses where `mode` is `ReadMode.STORED`.
+
+Block links in payload content will only include the target block ID.
+
+## ReadMode
+
+```proto
+enum ReadMode {
+  DEFAULT = 0;
+  STORED = 1;
+  LOGGED = 2;
+}
+```
+
+Block create / read / write commands may specify the subset of fields they wish to receive in the response
+by using the `ReadMode mode` field in the request. \
+The mode signals the intent. It is up to the individual block to determine which fields match the mode.
+Blocks are not guarantee to return the same subset of fields for each read request with the same mode.
```

Comment:
this sounds like mode doesn't deterministically determines which fields are included

Reply:
It does not. That's the point.